src/libraries/System.Data.OleDb/src/OleDbStruct.cs

```diff
@@ -33,11 +33,35 @@ internal enum DBBindStatus
     }
 #endif
 
-#if (WIN32 && !ARCH_arm)
```

Comment:
I see the same ifdef on many other structs in this file. Do all of them need to be fixed?

Reply:
Either that or we shoukd remove the attributes/defines for them

Reply:
The other structs will need to be fixed as well. I can take those up in another PR. I vote for this particular fix to be merged as this has been reported by users and can be caught in the tests as well (while running on x86 platform). The tests are not catching the other cases, so I think there may be more tests needed to validate the changes to the other structs here.

Reply:
I am fine with doing just a baby step if we believe that this library is useful for real workloads on x86 without fixing all/most of these. What are the chances that customers hitting this crash will immediately hit the next once once this one is fixed? Do we have some more complex sample app that uses this library that we can use to validate that it can work for a real app on x86?